tss-esapi/tests/integration_tests/abstraction_tests/credential_tests.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2025 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use tss_esapi::{
+    abstraction::{ak, ek, AsymmetricAlgorithmSelection},
+    attributes::SessionAttributesBuilder,
+    constants::SessionType,
+    handles::AuthHandle,
+    interface_types::{
+        algorithm::{HashingAlgorithm, SignatureSchemeAlgorithm},
+        ecc::EccCurve,
+        session_handles::PolicySession,
+    },
+    structures::{Digest, SymmetricDefinition},
+    utils,
+};
+
+use elliptic_curve::PublicKey;
+
+use crate::common::create_ctx_without_session;
+
+#[test]
+fn test_credential() {
+    let mut context = create_ctx_without_session();
+
+    let ek_ecc = ek::create_ek_object(
+        &mut context,
+        AsymmetricAlgorithmSelection::Ecc(EccCurve::NistP256),
+        None,
+    )
+    .unwrap();
+
+    let (ek_pub, _, _) = context.read_public(ek_ecc).unwrap();
+
+    let ak_res = ak::create_ak(
+        &mut context,
+        ek_ecc,
+        HashingAlgorithm::Sha384,
+        AsymmetricAlgorithmSelection::Ecc(EccCurve::NistP384),
+        SignatureSchemeAlgorithm::EcDsa,
+        None,
+        None,
+    )
+    .unwrap();
+
+    let ak_ecc = ak::load_ak(
+        &mut context,
+        ek_ecc,
+        None,
+        ak_res.out_private,
+        ak_res.out_public,
+    )
+    .unwrap();
+
+    let (_, key_name, _) = context.read_public(ak_ecc).unwrap();
+    let cred = vec![1, 2, 3, 4, 5];
+    let expected = Digest::try_from(vec![1, 2, 3, 4, 5]).unwrap();
+
+    let (credential_blob, secret) = utils::make_credential_ecc(
+        PublicKey::<p256::NistP256>::try_from(&ek_pub).unwrap(),
+        &cred,
+        key_name,
+    );
+
+    println!("{credential_blob:?}");
+    println!("{secret:?}");
+
+    let (session_aastributes, session_attributes_mask) = SessionAttributesBuilder::new().build();
```

Comment:
`session_aastributes` -> `session_attributes`

Reply:
This one was a copy paste from the other credentials tests. I wasn't sure if there was a special meaning or it was a typo.

Reply:
https://github.com/parallaxsecond/rust-tss-esapi/pull/564